### PR TITLE
fix(intervals): sort ASC

### PR DESCRIPTION
**Pull request by imwh0im**
- #3614

### CODE REVIEW

File: src/areIntervalsOverlapping/test.ts [hunk 1]
```diff
@@ -142,6 +142,14 @@ describe("areIntervalsOverlapping", () => {
     assert(isOverlapping);
   });
 
+  it('accepts timestamp 2', () => {
```
Comment (imwh0im) [inline src/areIntervalsOverlapping/test.ts:145]: Needs Rename

**Comment by imwh0im** [comment]
```js
const a = +new Date('1970-01-01T02:00:00.000Z');
const b = +new Date('1970-01-01T03:00:00.000Z');
const c = +new Date('1969-12-31T23:30:00.000Z');
const d = +new Date('1970-01-01T02:30:00.000Z');

console.log(a, b, c, d);
console.log([a, b].sort(), [c, d].sort())
console.log(
  [a, b].sort((a, b) => a - b),
  [c, d].sort((a, b) => a - b),
);
```

```bash
> 7200000 10800000 -1800000 9000000
> Array [10800000, 7200000] Array [-1800000, 9000000]
> Array [7200000, 10800000] Array [-1800000, 9000000
```

**Review by kossnocorp — APPROVED** [review]
Thank you a lot!

**Comment by kossnocorp** [comment]
Shipped with `date-fns@3.0.6`

**Comment by Azmat7860** [comment]
When I update the this package from v2.30.0 to v3.0.6 (latest version) then it gives this error 
./node_modules/date-fns/parse/_lib/parsers.mjs 
Can't import the named export 'AMPMMidnightParser' from non EcmaScript module (only default export is available)
Please look at this error or if anyone have solution of this error please provide here

**Pull request merged**